core/dbt/events/types.py

```diff
@@ -636,15 +638,210 @@ class InvalidDisabledSourceInTestNode(WarnLevel, CliEventABC):
+        stats_line = ("\nDone. PASS={pass} WARN={warn} ERROR={error} SKIP={skip} TOTAL={total}")
+        return stats_line.format(**self.stats)
+
+
+@dataclass
+class RunResultError(ErrorLevel, CliEventABC):
+    msg: str
+
+    def cli_msg(self) -> str:
+        return f"  {self.message}"
+
+
+@dataclass
+class RunResultErrorNoMessage(ErrorLevel, CliEventABC):
+    status: str
+
+    def cli_msg(self) -> str:
+        return f"  Status: {self.status}"
+
+
+@dataclass
+class SQLCompiledPath(InfoLevel, CliEventABC):
+    path: str
+
+    def cli_msg(self) -> str:
+        return f"  compiled SQL at {self.path}"
+
+
+@dataclass
+class CheckNodeTestFailure(InfoLevel, CliEventABC):
+    relation_name: str
+
+    def cli_msg(self) -> str:
+        msg = f"select * from {self.relation_name}"
+        border = '-' * len(msg)
+        return f"  See test failures:\n  {border}\n  {msg}\n  {border}"
+
+
+@dataclass
+class FirstRunResultError(ErrorLevel, CliEventABC):
+    msg: str
+
+    def cli_msg(self) -> str:
+        return ui.yellow(self.msg)
+
+
+@dataclass
+class AfterFirstRunResultError(ErrorLevel, CliEventABC):
+    msg: str
+
+    def cli_msg(self) -> str:
+        return self.msg
+
+
+@dataclass
+class EndOfRunSummary(InfoLevel, CliEventABC):
+    num_errors: int
+    num_warnings: int
+    keyboard_interrupt: bool = False
+
+    def cli_msg(self) -> str:
+        error_plural = utils.pluralize(self.num_errors, 'error')
+        warn_plural = utils.pluralize(self.num_warnings, 'warning')
+        if self.keyboard_interrupt:
+            message = ui.yellow('Exited because of keyboard interrupt.')
+        elif self.num_errors > 0:
+            message = ui.red("Completed with {} and {}:".format(
+                error_plural, warn_plural))
+        elif self.num_warnings > 0:
+            message = ui.yellow('Completed with {}:'.format(warn_plural))
+        else:
+            message = ui.green('Completed successfully')
+        return message
+
+
+def format_fancy_output_line(
+        msg: str, status: str, index: Optional[int],
+        total: Optional[int], execution_time: Optional[float] = None,
+        truncate: bool = False
+) -> None:
+    if index is None or total is None:
+        progress = ''
+    else:
+        progress = '{} of {} '.format(index, total)
+    prefix = "{timestamp} | {progress}{message}".format(
+        timestamp=get_timestamp(),
+        progress=progress,
+        message=msg)
+
+    truncate_width = ui.printer_width() - 3
+    justified = prefix.ljust(ui.printer_width(), ".")
+    if truncate and len(justified) > truncate_width:
+        justified = justified[:truncate_width] + '...'
+
+    if execution_time is None:
+        status_time = ""
+    else:
+        status_time = " in {execution_time:0.2f}s".format(
+            execution_time=execution_time)
+
+    output = "{justified} [{status}{status_time}]".format(
+        justified=justified, status=status, status_time=status_time)
+
+    return output
```

Comment:
Thoughts on this living here like this?

Reply:
I'm fine with this. I would also be fine with it living in another module and we just import it for now till we can migrate it in elegantly. I actually much prefer when a lot of this logic is in functions and colocated with the messages themselves.